src/main/java/com/squareup/javapoet/Util.java

```diff
@@ -106,7 +106,45 @@ public static void requireExactlyOneOf(Set<Modifier> modifiers, Modifier... mutu
         modifiers, Arrays.toString(mutuallyExclusive));
   }
 
-  public static boolean hasDefaultModifier(Collection<Modifier> modifiers) {
+  static boolean hasDefaultModifier(Collection<Modifier> modifiers) {
     return DEFAULT != null && modifiers.contains(DEFAULT);
   }
+
+  // {@linkplain https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.10.6}
```

Comment:
This isn't Javadoc so the `{@linkplain` wrapper isn't needed. But more importantly, I would put this comment inside the method so that if it ever gets moved by automated refactoring tools the comment moves with it.